dotnet/src/Microsoft.AutoGen/Extensions/AIModelClientHostingExtensions/ServiceCollectionChatCompletionExtensions.cs

```diff
@@ -0,0 +1,117 @@
+        var endpoint = (string?)connectionStringBuilder["endpoint"];
+        var apiKey = (string)connectionStringBuilder["key"] ?? throw new InvalidOperationException($"The connection string named '{serviceName}' does not specify a value for 'Key', but this is required.");
+
+        modelOrDeploymentName ??= (connectionStringBuilder["Deployment"] ?? connectionStringBuilder["Model"]) as string;
+        if (string.IsNullOrWhiteSpace(modelOrDeploymentName))
+        {
+            throw new InvalidOperationException($"The connection string named '{serviceName}' does not specify a value for 'Deployment' or 'Model', and no value was passed for {nameof(modelOrDeploymentName)}.");
+        }
+
+        var endpointUri = string.IsNullOrEmpty(endpoint) ? null : new Uri(endpoint);
+        return hostBuilder.Services.AddOpenAIChatClient(apiKey, modelOrDeploymentName, endpointUri, builder);
+    }
+    public static IServiceCollection AddOpenAIChatClient(
+        this IServiceCollection services,
+        string apiKey,
+        string modelOrDeploymentName,
+        Uri? endpoint = null,
+        Func<ChatClientBuilder, ChatClientBuilder>? builder = null)
+    {
+        return services
+            .AddSingleton(_ => endpoint is null
+                ? new OpenAIClient(apiKey)
+                : new AzureOpenAIClient(endpoint, new ApiKeyCredential(apiKey)))
+            .AddChatClient(pipeline =>
+            {
+                builder?.Invoke(pipeline);
+                var openAiClient = pipeline.Services.GetRequiredService<OpenAIClient>();
+                return pipeline.Use(openAiClient.AsChatClient(modelOrDeploymentName));
+            });
+    }
+    public static IServiceCollection AddAzureChatClient(
+        this IHostApplicationBuilder hostBuilder,
+        string serviceName,
+        Func<ChatClientBuilder, ChatClientBuilder>? builder = null,
+        string? modelOrDeploymentName = null)
+    {
+        if (modelOrDeploymentName is null)
+        {
+            var configKey = $"{serviceName}:LlmModelName";
+            modelOrDeploymentName = hostBuilder.Configuration[configKey];
+            if (string.IsNullOrEmpty(modelOrDeploymentName))
+            {
+                throw new InvalidOperationException($"No {nameof(modelOrDeploymentName)} was specified, and none could be found from configuration at '{configKey}'");
+            }
+        }
+        var endpoint = $"{serviceName}:Endpoint" ?? throw new InvalidOperationException($"No endpoint was specified for the Azure Inference Chat Client");
+        var endpointUri = string.IsNullOrEmpty(endpoint) ? null : new Uri(endpoint);
+        return hostBuilder.Services.AddChatClient(pipeline =>
+        {
+            builder?.Invoke(pipeline);
+            var token = Environment.GetEnvironmentVariable("GH_TOKEN") ?? throw new InvalidOperationException("No model access token was found in the environment variable GH_TOKEN");
```

Comment:
How does this one work? Does it actually come from the EnvironmentVariable or should it be passed in as a param?